packages/rest/test/unit/router/trie-router.unit.ts

```diff
@@ -36,30 +46,76 @@ describe('trie router', () => {
       {verb: 'get', path: '/orders/{id}/exists'},
       {verb: 'get', path: '/orders/{id}'},
       {verb: 'delete', path: '/orders/{id}'},
+      {verb: 'get', path: '/users/{id}'},
+      {verb: 'get', path: '/users/{userId}/orders'},
+      {verb: 'get', path: '/users/{id}/products'},
     ]) {
       expect(staticRoutes).to.not.containEql(r);
     }
   });
 
-  it('list routes by order', () => {
-    const router = givenTrieRouter();
-
+  it('lists routes by order', () => {
     expect(router.list().map(getVerbAndPath)).to.eql([
-      {verb: 'post', path: '/orders'},
+      {verb: 'get', path: '/orders/count'},
+      {verb: 'get', path: '/orders/{id}/exists'},
       {verb: 'put', path: '/orders/{id}'},
       {verb: 'patch', path: '/orders/{id}'},
-      {verb: 'patch', path: '/orders'},
-      {verb: 'get', path: '/orders/{id}/exists'},
-      {verb: 'get', path: '/orders/count'},
       {verb: 'get', path: '/orders/{id}'},
-      {verb: 'get', path: '/orders'},
       {verb: 'delete', path: '/orders/{id}'},
+      {verb: 'post', path: '/orders'},
+      {verb: 'patch', path: '/orders'},
+      {verb: 'get', path: '/orders'},
       {verb: 'delete', path: '/orders'},
+      {verb: 'get', path: '/users/{userId}/orders'},
+      {verb: 'get', path: '/users/{id}/products'},
+      {verb: 'get', path: '/users/{id}'},
     ]);
   });
 
+  it('finds route for GET /users/{id}', () => {
+    const req = givenRequest({method: 'get', url: '/users/123'});
+    const route = router.find(req);
+    expect(getRouteInfo(route)).to.containEql({
+      verb: 'get',
+      path: '/users/{id}',
+      params: {id: '123'},
+    });
+  });
+
+  it('finds route for GET /users/{id}/products', () => {
+    const req = givenRequest({method: 'get', url: '/users/123/products'});
+    const route = router.find(req);
+    expect(getRouteInfo(route)).to.containEql({
+      verb: 'get',
+      path: '/users/{id}/products',
+      params: {id: '123'},
+    });
+  });
+
+  it('finds route for GET /users/{userId}/orders', () => {
```

Comment:
I find these two tests ("finds route for GET /users/{id}/products" and "finds route for GET /users/{userId}/orders") very difficult to understand. The paths looks pretty much the same, why would one test fail when the other is passing?

Please find a way how to capture the problem you are solving in the structure of the test suite. For example, the test can make it more clear that parameterized paths using different path parameter names are correctly "merged" together and point endpoints work as expected.

It would be also great to move the setup of routes closer to the tests. Right now it's not clear at all which test is relying on which routes defined by `givenRoutes`.

A good test  uses only the minimal setup is strictly required by the tested scenario. By creating a new test setup razor-focused on needs of each test case, the tests become easier to understand, since it’s immediately clear what parts of the setup are relevant to the tests. If the tests share a large setup with many routes, it is difficult to tell which routes are actually relevant to the tested scenario.